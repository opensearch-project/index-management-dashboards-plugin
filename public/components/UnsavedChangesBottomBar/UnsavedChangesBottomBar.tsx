/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useEffect, useRef, useState } from "react";
import { EuiButton, EuiBottomBar, EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from "@elastic/eui";

export type CustomFormRowProps = {
  unsavedCount: number;
  onClickCancel: () => Promise<void>;
  onClickSubmit: () => Promise<void>;
  submitButtonDataTestSubj?: string;
};

export default function CustomFormRow(props: CustomFormRowProps) {
  const { unsavedCount, onClickCancel, onClickSubmit, submitButtonDataTestSubj } = props;
  const [loading, setLoading] = useState(false);
  const bottomBarRef = useRef(null);
  const destroyRef = useRef(false);
  const onClick = async () => {
    setLoading(true);
    try {
      await onClickSubmit();
    } catch (e) {
    } finally {
      if (destroyRef.current) {
        return;
      }
      setLoading(false);
    }
  };

  useEffect(() => {
    const bodyDom = document.querySelector<HTMLDivElement>("#opensearch-dashboards-body");
    let originalBodyPaddingBottom = "";
    if (bodyDom) {
      originalBodyPaddingBottom = bodyDom.style.paddingBottom;
      bodyDom.style.paddingBottom = "64px";
    }

    return () => {
      destroyRef.current = true;
      if (bodyDom) {
        bodyDom.style.paddingBottom = originalBodyPaddingBottom;
      }
    };
  }, []);

  return (
    <EuiBottomBar ref={bottomBarRef}>
      <EuiFlexGroup alignItems="center">
        <EuiFlexItem>{unsavedCount} unsaved changes.</EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty onClick={onClickCancel} color="ghost" iconType="cross" size="s">
            Cancel changes
          </EuiButtonEmpty>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButton
            data-test-subj={submitButtonDataTestSubj}
            onClick={onClick}
            isLoading={loading}
            disabled={loading}
            iconType="check"
            color="secondary"
            fill
            size="s"
          >
            Save changes
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiBottomBar>
  );
}
