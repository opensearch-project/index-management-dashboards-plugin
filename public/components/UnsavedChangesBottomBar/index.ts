import UnsavedChangesBottomBar from "./UnsavedChangesBottomBar";

export default UnsavedChangesBottomBar;
