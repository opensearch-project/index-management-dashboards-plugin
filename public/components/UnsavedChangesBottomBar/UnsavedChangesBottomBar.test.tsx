/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import UnsavedChangesBottomBar from "./index";

describe("<FormGenerator /> spec", () => {
  it("render the component", () => {
    render(<UnsavedChangesBottomBar unsavedCount={0} onClickCancel={async () => {}} onClickSubmit={async () => {}} />);
    expect(document.body.children).toMatchSnapshot();
  });
});
