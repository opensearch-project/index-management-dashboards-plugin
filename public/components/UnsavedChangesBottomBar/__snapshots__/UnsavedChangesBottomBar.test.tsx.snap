// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGenerator /> spec render the component 1`] = `
HTMLCollection [
  <div />,
  <div>
    <section
      aria-label="Page level controls"
      class="euiBottomBar euiBottomBar--fixed euiBottomBar--paddingMedium"
      style="left: 0px; right: 0px; bottom: 0px;"
    >
      <h2
        class="euiScreenReaderOnly"
      >
        Page level controls
      </h2>
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          0
           unsaved changes.
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--ghost euiButtonEmpty--small"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              EuiIconMock
              <span
                class="euiButtonEmpty__text"
              >
                Cancel changes
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--secondary euiButton--small euiButton--fill"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              EuiIconMock
              <span
                class="euiButton__text"
              >
                Save changes
              </span>
            </span>
          </button>
        </div>
      </div>
    </section>
    <p
      aria-live="assertive"
      class="euiScreenReaderOnly"
    >
      There is a new region landmark with page level controls at the end of the document.
    </p>
  </div>,
]
`;
